test/threadstest.c

```diff
@@ -654,6 +654,19 @@ static int test_obj_add(void)
                            1, default_provider);
 }
 
+static void test_lib_ctx_load_config_worker(void)
+{
+    if (!TEST_int_eq(OSSL_LIB_CTX_load_config(multi_libctx, config_file), 1))
+        multi_success = 0;
+}
+
+static int test_lib_ctx_load_config(void)
+{
+    return thread_run_test(&test_lib_ctx_load_config_worker,
+                           MAXIMUM_THREADS, &test_lib_ctx_load_config_worker,
+                           1, default_provider);
+}
+
```

Comment:
It would be interesting to know whether this test without the fix in this PR would have caught the problem.

Reply:
https://github.com/openssl/openssl/pull/18405

Caught! Thank you tsan.